### DISCUSSION TITLE
Задача

### DISCUSSION THREAD

**Discussion by antontelichenko** [HowProgrammingWorks/LiveQA#30] [Q&A] [answered]
![photo_2022-04-07 15 59 41](https://user-images.githubusercontent.com/33400238/162204600-0725547e-768b-40fd-ae8d-b88459f63324.jpeg)

интересует ответ

**Comment by Semigradsky** [comment]
Достаточно набрать в консоли браузера, чтобы узнать, разве нет?
Если считать в уме, то удобнее по частям "выполнять".

**Reply by tshemsedinov** [reply]
Я думаю, что интересует решение задачи - а задача была - исправить ошибку неконсистентности типов в коде.

**Comment by antontelichenko** [comment]
Да, интересует как исправить
Чт, 7 апр. 2022 г. в 19:51, Timur Shemsedinov ***@***.***>:

**Comment by tshemsedinov** [comment, marked as answer]
@geka412315 @Semigradsky  Разберем код: по сути мы сделали тип `coord` и на вход передаем два экземпляра наследников этого типа `x` и `offset`, а на выход получаем не `coord`, а `Number`, т.е. складывали метры с метрами, а получили килограммы ))) Для исправления этой ситуации, нужно обернуть выражение `x1 + x2` в тип `coord` и в результате выйдет: `resolve(coord()(x1 + x2))`

**Comment by antontelichenko** [comment]
Я думал что ответ coord()(x1+Y2), тк coord это функция которая ничего не
принимает и при вызове возвращает функцию, в которую уже можно закинуть
x1,y2. 🤔

Чт, 7 апр. 2022 г. в 20:06, Timur Shemsedinov ***@***.***>:

**Reply by tshemsedinov** [reply]
Да, точно, поправил, спасибо. Желательно отвечать в гитхабе, а то почтовый шлю из каждого письма делает отдельный ответ )

**Reply by antontelichenko** [reply]
не было возможности на гитхаб зайти, только по почте мог ответить, мог и подождать, но было слишком интересно правильно-ли сделал)))
